app/src/main/java/com/jrod7938/textchangeapp/screens/search/BookSearchScreen.kt

```diff
@@ -1,47 +1,278 @@
+                            modifier = Modifier.padding(end = 15.dp)
+                        )
+                        Text(text = it)
+                    }
+                }
+            }
+
+        }
+
+        // Search Filter Title Text
+
+        Row(horizontalArrangement = Arrangement.Start,
+            modifier = Modifier
+                .padding(top = 15.dp,
+                bottom = 15.dp,
+                start = 30.dp)){
+            Text("Search Options")
+            Spacer(modifier = Modifier.padding(5.dp))
+            ClickableText(
+                text = AnnotatedString(text = if(filterBarActive) "Show" else "Hide"), onClick = {
+                    filterBarActive = !filterBarActive
+                    filter = SearchType.ISBN},
+                style = TextStyle(color = MaterialTheme.colorScheme.primary,
+                    textDecoration = TextDecoration.Underline  )
+            )
+        }
+
+        // Bar
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center,
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            AnimatedVisibility(visible = a,
+                enter = slideInVertically(
+                    initialOffsetY = { -40 }
+                ) + fadeIn(initialAlpha = 0.3f),
+                exit = slideOutVertically() + fadeOut()) {
+
+
+                val options = arrayOf(
+                    ToggleButtonOption("ISBN", iconRes = null),
+                    ToggleButtonOption("Title", iconRes = null),
+                    ToggleButtonOption("Author", iconRes = null)
+                )
+
+                ToggleButton(
+                    options = options,
+                    type = SelectionType.SINGLE,
+                    modifier = Modifier
+                        .padding(end = 4.dp)
+                        .semantics { contentDescription = "Toggle Button"}
+                ) { selectedOption ->
+                    filter = when (selectedOption[0].text) {
+                        "ISBN" -> SearchType.ISBN
+                        "Title" -> SearchType.Title
+                        "Author" -> SearchType.Author
+                        else -> SearchType.ISBN
+                    }
+                }
+            }
+        }
+
+
+        LaunchedEffect(filterBarActive) {
+            launch {
+                a = !a
+            }
+        }
+
+        Column(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            if (!errorMessage.isNullOrEmpty()) {
+                Text(text = "$errorMessage")
+            }
+            if (loading) {
+                CircularProgressIndicator()
+            }
+        }
+
+        AnimatedVisibility(visible = !loading && onSearchClicked) {
+            DisplaySearchResults(bookList, text, navController = navController, filter = filter)

```

Comment:
When a book is added to favorites and the favorite button is pressed, the book is not removed from the database. Please update code to ensure the book is removed from the database when the book is unsaved

Reply:
All I did was reference the function written. Are you asking me to fix the unsave function? And where in the database does this need to be removed?

Reply:
No need, I'll look into this when implementing the navigation from Home screen Category to Search Screen